cmds/core/chmod/chmod_test.go

```diff
@@ -226,31 +231,29 @@ func TestChmod(t *testing.T) {
 			name:      "bad filepath",
 			args:      []string{"a=rx", "pathdoes not exist"},
 			recursive: true,
-			want:      "chmod pathdoes not exist: no such file or directory",
+			err:       os.ErrNotExist,
 		},
 		{
 			name:       "correct path filepath",
 			args:       []string{"0777", f.Name()},
 			recursive:  true,
 			modeBefore: 0o777,
 			modeAfter:  0o777,
-			want:       "chmod pathdoes not exist: no such file or directory",
+			err:        nil,
 		},
 	} {
 		t.Run(tt.name, func(t *testing.T) {
-			*recursive = tt.recursive
-			*reference = tt.reference
 			os.Chmod(f.Name(), tt.modeBefore)
-			mode, got := chmod(tt.args...)
-			if got != nil {
-				if got.Error() != tt.want {
-					t.Errorf("chmod() = %q, want: %q", got.Error(), tt.want)
-				}
-			} else {
-				if mode != tt.modeAfter {
-					t.Errorf("chmod() = %v, want: %v", mode, tt.modeAfter)
-				}
+			mode, err := chmod(tt.recursive, tt.reference, tt.args...)
+			t.Logf("chmod(%v, %q, %q) = %v, want %v", tt.recursive, tt.reference, tt.args, err, tt.err)
+			if !errors.Is(err, tt.err) {
+				t.Errorf("chmod(%v, %q, %q) = %v, want %v", tt.recursive, tt.reference, tt.args, err, tt.err)
```

Comment:
I think you can replace "t.Errorf(X); return"  with just t.Fatalf(X), right?

Reply:
no, not in a test like this. 

t.Errorf marks ONE test as failing, but it lets you run all the tests, so you can see which ones failed and which did not. 
In this case, only one test failed, and knowing that was useful.

That's why Errorf is there :-)